toast-er.html

```diff
@@ -308,22 +308,26 @@
 
         // Get configuration of the toast.
         var bread = this._breads[0],
-          duration = bread.duration || this.duration,
-          type = bread.type || this.type,
+          buttonEvent = bread.buttonEvent || this.buttonEvent,
+          buttonText = bread.buttonText || this.buttonText,
+          hasButton = buttonText.length != 0,
+          duration = bread.duration || (hasButton ? 10000 : this.duration),
```

Comment:
@wincinderith Or simply leave the `duration` like before:

```javascript
duration = bread.duration || this.duration
```

And replace the line 326 by:

```javascript
toast.duration = (hasButton && duration < 10000) ? 10000 : duration;
```

❓